src/LRS.js

```diff
@@ -542,6 +609,46 @@ TinCan client library
         },
 
         /**
+        Assigns attachment content to the correct attachment to create a StatementsResult object that is sent
+        to the callback of queryStatements()
+
+        @method _assignAttachmentContent
+        @private
+        @param {Array} Array of TinCan Statement JSON objects
+        @param {Array} Array containing lines from the HTTP response
+        @return {Array} Array of TinCan Statement JSON objects with correctly assigned attachment content
+        */
+        _assignAttachmentContent: function (stmts, lines) {
+            var hashToMatch,
+                content,
+                i,
+                x,
+                a;
+
+            for (i = 0; i < lines.length; i += 1) {
+                if (lines[i].startsWith("X-Experience-API-Hash")) {
+                    hashToMatch = lines[i].split(":")[1];
```

Comment:
Not sure this is sufficient to parse headers. Can we avoid parsing the headers ourselves? At the very least this probably breaks on whitespace.